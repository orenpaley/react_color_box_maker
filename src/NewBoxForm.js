import React, { useState } from "react";
import { v4 as uuid } from "uuid";

const NewBoxForm = ({ addBox }) => {
  const INITIAL_STATE = {
    height: 250,
    width: 250,
    backgroundColor: "green",
  };
  const [formData, setFormData] = useState(INITIAL_STATE);
  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    addBox({ ...formData, key: uuid() });
    setFormData(INITIAL_STATE);
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="height">Height</label>
      <input
        id="height"
        type="range"
        name="height"
        min="25"
        max="500"
        step={50}
        placeholder="Box Height"
        value={formData.height}
        onChange={handleChange}
      />
      <label htmlFor="width">Width</label>
      <input
        id="width"
        type="range"
        name="width"
        min="25"
        max="500"
        step={50}
        value={formData.width}
        onChange={handleChange}
      />
      <label htmlFor="backgroundColor">Color</label>
      <input
        id="backgroundColor"
        type="text"
        name="backgroundColor"
        value={formData.backgroundColor}
        onChange={handleChange}
      />
      <button>Add Box</button>
    </form>
  );
};

export default NewBoxForm;
