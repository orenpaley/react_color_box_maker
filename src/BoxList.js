import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import { useState } from "react";
import { v4 as uuid } from "uuid";

const BoxList = () => {
  const INITIAL_STATE = [
    { height: 250, width: 250, backgroundColor: "red", id: uuid() },
    { height: 150, width: 350, backgroundColor: "orange", id: uuid() },
    { height: 350, width: 150, backgroundColor: "green", id: uuid() },
  ];
  const [Boxes, SetBoxes] = useState(INITIAL_STATE);
  const addBox = (newBox) => {
    SetBoxes((boxes) => [...boxes, { ...newBox, id: uuid() }]);
  };
  const removeBox = (boxToRemove) => {
    SetBoxes((Boxes) => Boxes.filter((b) => b.id !== boxToRemove.id));
  };
  return (
    <div>
      <h1>Box Maker</h1>
      <NewBoxForm addBox={addBox} />
      <div className="BoxList">
        {Boxes.map(({ width, height, backgroundColor, id }) => (
          <Box
            key={id}
            id={id}
            width={width}
            height={height}
            backgroundColor={backgroundColor}
            remove={removeBox}
          />
        ))}
      </div>
    </div>
  );
};

export default BoxList;
