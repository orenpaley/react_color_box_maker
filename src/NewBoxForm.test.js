import NewBoxForm from "./NewBoxForm";
import { Boxes, addBox } from "./BoxList";

import { fireEvent, render } from "@testing-library/react";
import { toBePartiallyChecked } from "@testing-library/jest-dom/dist/matchers";

// test("renders learn react link", () => {
//   render(<BoxLisst />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

it("renders without crashing", () => {
  render(<NewBoxForm addBox={addBox} />);
});
it("matches snapshot", function () {
  const { asFragment } = render(<NewBoxForm addBox={addBox} />);
  expect(asFragment()).toMatchSnapshot();
});

it("changes values of sliders", () => {
  const { getByLabelText, queryByText } = render(
    <NewBoxForm addBox={addBox} />
  );
  let height = getByLabelText("Height");
  let width = getByLabelText("Width");
  let color = getByLabelText("Color");
  const btn = queryByText("Add Box");
  expect(400).not.toBeInTheDocument();
  fireEvent.change(height, { target: { value: 400 } });
  expect(400).toBeInTheDocument();
  expect(500).not.toBeInTheDocument();
  fireEvent.change(width, { target: { value: 500 } });
  expect(500).toBeInTheDocument();
  fireEvent.change(color, { target: { value: "pink" } });
  expext(Boxes).length.not.toBe(4);
  fireEvent.click(btn);
  expext(Boxes).length.toBe(4);
});
