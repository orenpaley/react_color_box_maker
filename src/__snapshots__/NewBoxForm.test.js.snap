// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      max="500"
      min="25"
      name="height"
      placeholder="Box Height"
      step="50"
      type="range"
      value="250"
    />
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      max="500"
      min="25"
      name="width"
      step="50"
      type="range"
      value="250"
    />
    <label
      for="backgroundColor"
    >
      Color
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      type="text"
      value="green"
    />
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
